build_tools/github/build_doc.sh

```diff
@@ -20,7 +20,8 @@ set -e
 if [ -n "$GITHUB_ACTION" ]
 then
     # Map the variables for the new documentation builder to the old one
-    CIRCLE_SHA1=$(git log --no-merges -1 --pretty=format:%H)
+    CIRCLE_SHA1=$(git log -1 --pretty=format:%H)
```

Comment:
Removing the `--no-merges` fixes the issue with finding the changes.

**Note**: I tried using `GITHUB_SHA` here: [`8de2674` (#23534)](https://github.com/scikit-learn/scikit-learn/pull/23534/commits/8de2674bbfec3cc08af79d855d428d1ccb9f06ed), but it didn't pick up the diff.